lib/events/index.js

```diff
@@ -2,30 +2,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var url = require('url');
-var Promise = require('bluebird');
-var basket = require('../basket');
-var logger = require('../logging')('events');
+const url = require('url');
+const Promise = require('bluebird');
+const basket = require('../basket');
+const logger = require('../logging')('events');
 
-var config = require('../config');
+const config = require('../config');
 
 // Certain special values of the utm_campaign metrics parameters
 // are used to indicate a newsletter campaign, and cause us to
 // auto-subscribe the user to a particular newsletter.
-var NEWSLETTER_CAMPAIGNS = config.get('basket.newsletter_campaigns');
-var NEWSLETTER_ID_REGISTER = config.get('basket.newsletter_id_register');
-var SOURCE_URL_BASE = config.get('basket.source_url');
+const NEWSLETTER_CAMPAIGNS = config.get('basket.newsletter_campaigns');
+const NEWSLETTER_ID_REGISTER = config.get('basket.newsletter_id_register');
+const SOURCE_URL_BASE = config.get('basket.source_url');
 
 
-var messageHandlers = {
+const messageHandlers = module.exports._messageHandlers = {
   verified: onVerified,
   login: onLogin
 };
 
+const DISABLED_EVENT_TYPES = config.get('basket.sqs.disabled_event_types');
+if (DISABLED_EVENT_TYPES) {
+  DISABLED_EVENT_TYPES.forEach((typ) => {
+    delete messageHandlers[typ];
```

Comment:
I'll withhold my typical objection about  `delete` here because it's just initialisation code.

Reply:
I'm open to doing this a different way, e.g. we could have one that's the set of all handlers, but copy the non-disabled ones into a separate object for use at runtime.

Reply:
Or even just setting to `undefined`, I prefer to `delete` generally. But it's initialisation code, it only gets executed once, so whatever. :smile: